In [397]:
import json
import pandas as pd
import matplotlib as plt

Amir Wilson's Instagram Data Report
September 13, 2024
I downloaded this data but I did not necessarily create it, it is more of an assessment and collection of information on how I used the platform "Instagram"
This data was created to help Meta better understand its users or even to help them with sponsorships or advertisements. It is available to me to help me better understand myself or even for a sense of transparency for the company to the user.
This data can be reliable because a lot of the data can be verified once shown however, it could be unreliable because there could be data being withheld. 

In [399]:
with open(r"liked_posts.json") as j:
    dict = json.load(j)

In [400]:
df = pd.DataFrame(dict)

In [401]:
df.head()

likes_media_likes
0  {'title': 'sixers', 'string_list_data': [{'hre...
1  {'title': 'jmccain24', 'string_list_data': [{'...
2  {'title': 'deendose', 'string_list_data': [{'h...
3  {'title': 'espn', 'string_list_data': [{'href'...
4  {'title': 'zipbby', 'string_list_data': [{'hre...

In [402]:
dict.keys()

dict_keys(['likes_media_likes'])

In [403]:
df2 = pd.DataFrame(dict['likes_media_likes'])

In [404]:
df2.head()

title                                   string_list_data
0     sixers  [{'href': 'https://www.instagram.com/p/C_taAUV...
1  jmccain24  [{'href': 'https://www.instagram.com/reel/C_tH...
2   deendose  [{'href': 'https://www.instagram.com/p/C_tpYp0...
3       espn  [{'href': 'https://www.instagram.com/p/C_t3WF7...
4     zipbby  [{'href': 'https://www.instagram.com/p/C_twp48...

In [405]:
type(df2['string_list_data'].iloc[0])

list

In [406]:
len(df2['string_list_data'].iloc[0])

1

In [407]:
df2['string_list_data'].iloc[0][0]

{'href': 'https://www.instagram.com/p/C_taAUVR_Lp/',
 'value': 'ð\x9f\x91\x8d',
 'timestamp': 1725980331}

In [408]:
df2['string_list_data'].iloc[0][0]['timestamp']

1725980331

In [409]:
df2['Time of Like'] = [x[0]['timestamp'] for x in df2['string_list_data']]

In [410]:
df2.head()

title                                   string_list_data  Time of Like
0     sixers  [{'href': 'https://www.instagram.com/p/C_taAUV...    1725980331
1  jmccain24  [{'href': 'https://www.instagram.com/reel/C_tH...    1725929936
2   deendose  [{'href': 'https://www.instagram.com/p/C_tpYp0...    1725929507
3       espn  [{'href': 'https://www.instagram.com/p/C_t3WF7...    1725929081
4     zipbby  [{'href': 'https://www.instagram.com/p/C_twp48...    1725928816

In [411]:
df3 = pd.DataFrame([x[0] for x in df2['string_list_data']])

In [412]:
df3['title'] = df2['title']
df3.head()

href value   timestamp      title
0     https://www.instagram.com/p/C_taAUVR_Lp/  👍  1725980331     sixers
1  https://www.instagram.com/reel/C_tHOjPM1e-/  👍  1725929936  jmccain24
2     https://www.instagram.com/p/C_tpYp0P4wI/  👍  1725929507   deendose
3     https://www.instagram.com/p/C_t3WF7RBJ8/  👍  1725929081       espn
4     https://www.instagram.com/p/C_twp48yhb1/  👍  1725928816     zipbby

In [413]:
df3.groupby('title').count().sort_values('href', ascending = False).head(10)

href  value  timestamp
title                                       
wwe                    134    134        134
sportscenter           133    133        133
bleacherreport         125    125        125
sixers                  90     90         90
espn                    83     83         83
nba.in_depth            75     75         75
sidelinesources         73     73         73
futurecannibisfarmer    55     55         55
nba                     53     53         53
hidden.ny               48     48         48

The next thing I thought would be interesting to get a better look at regarding my data was my following. How many accounts have I followed, when did I follow them, and what types of accounts am I following, these are all questions that came to my mind. Knowing me and my behaviors on the app recently a hypothesis I wanted to test and thought that I could test given my current knowledge was that I have followed more people recently than I normally do or that I have before. This could be tested using the timestamp which I was familiar with from the first portion of the project and that fortified my interest in testing this hypothesis.

In [414]:
with open(r"following.json") as k:
    dict = json.load(k)

Opening the following json file

In [415]:
df4 = pd.DataFrame(dict)

In [416]:
df4.head()

relationships_following
0  {'title': '', 'media_list_data': [], 'string_l...
1  {'title': '', 'media_list_data': [], 'string_l...
2  {'title': '', 'media_list_data': [], 'string_l...
3  {'title': '', 'media_list_data': [], 'string_l...
4  {'title': '', 'media_list_data': [], 'string_l...

I was interested in seeing what a basic dataframe would look like before I looked at the keys or did any work on it.

In [417]:
dict.keys()

dict_keys(['relationships_following'])

In [418]:
df5 = pd.DataFrame(dict['relationships_following'])

In [419]:
df5.head(10)

title media_list_data                                   string_list_data
0                    []  [{'href': 'https://www.instagram.com/k1ng_vert...
1                    []  [{'href': 'https://www.instagram.com/swapafr',...
2                    []  [{'href': 'https://www.instagram.com/thankscho...
3                    []  [{'href': 'https://www.instagram.com/n.ieshhaa...
4                    []  [{'href': 'https://www.instagram.com/olarunt',...
5                    []  [{'href': 'https://www.instagram.com/phillygya...
6                    []  [{'href': 'https://www.instagram.com/nevergoin...
7                    []  [{'href': 'https://www.instagram.com/iwantnish...
8                    []  [{'href': 'https://www.instagram.com/foenemcre...
9                    []  [{'href': 'https://www.instagram.com/i1_7.c', ...

This did not make much sense to me but I saw instagram data and what seemed to be links to accounts which made me feel good and like I was progressing

In [420]:
type(df5['string_list_data'].iloc[0])

list

In [421]:
df5['string_list_data'].iloc[0][0]

{'href': 'https://www.instagram.com/k1ng_vert_',
 'value': 'k1ng_vert_',
 'timestamp': 1725887506}

Seeing what "string_list_data" held made me certain that what I wanted to test would be achievable it just was a matter of me having the information and knowledge to achieve it 

In [422]:
df5['Time of Follow'] = [x[0]['timestamp'] for x in df5['string_list_data']]

I wanted the time of follow to be clear so that it would be easier for me to test my hypothesis so I made time of follow the name of a column in the dataframe

In [423]:
df5.head(10)

title media_list_data                                   string_list_data  \
0                    []  [{'href': 'https://www.instagram.com/k1ng_vert...   
1                    []  [{'href': 'https://www.instagram.com/swapafr',...   
2                    []  [{'href': 'https://www.instagram.com/thankscho...   
3                    []  [{'href': 'https://www.instagram.com/n.ieshhaa...   
4                    []  [{'href': 'https://www.instagram.com/olarunt',...   
5                    []  [{'href': 'https://www.instagram.com/phillygya...   
6                    []  [{'href': 'https://www.instagram.com/nevergoin...   
7                    []  [{'href': 'https://www.instagram.com/iwantnish...   
8                    []  [{'href': 'https://www.instagram.com/foenemcre...   
9                    []  [{'href': 'https://www.instagram.com/i1_7.c', ...   

   Time of Follow  
0      1725887506  
1      1725679895  
2      1725642228  
3      1725386551  
4      1725262958  
5      1724616579  
6      1724426473  
7      1724284995  
8      1724256372  
9      1724060506

In [424]:
df5.shape

(596, 4)

Realized I didn't know the shape of the dataframe and I knew that would help me with my hypothesis

In [425]:
df5.columns

Index(['title', 'media_list_data', 'string_list_data', 'Time of Follow'], dtype='object')

In [426]:
del df5["media_list_data"]
del df5["title"]
df5.head()

string_list_data  Time of Follow
0  [{'href': 'https://www.instagram.com/k1ng_vert...      1725887506
1  [{'href': 'https://www.instagram.com/swapafr',...      1725679895
2  [{'href': 'https://www.instagram.com/thankscho...      1725642228
3  [{'href': 'https://www.instagram.com/n.ieshhaa...      1725386551
4  [{'href': 'https://www.instagram.com/olarunt',...      1725262958

Deleted some columns with the intention of making the dataframe easier to understand, I intended on adding more but title had nothing in it and media_list_data had an empty box symbol thing?

In [427]:
df5['Account Name'] = [x[0]['value'] for x in df5['string_list_data']]
df5.head()

string_list_data  Time of Follow  \
0  [{'href': 'https://www.instagram.com/k1ng_vert...      1725887506   
1  [{'href': 'https://www.instagram.com/swapafr',...      1725679895   
2  [{'href': 'https://www.instagram.com/thankscho...      1725642228   
3  [{'href': 'https://www.instagram.com/n.ieshhaa...      1725386551   
4  [{'href': 'https://www.instagram.com/olarunt',...      1725262958   

  Account Name  
0   k1ng_vert_  
1      swapafr  
2   thankschow  
3    n.ieshhaa  
4      olarunt

In [428]:
df5['Link to Account'] = [x[0]['href'] for x in df5['string_list_data']]
df5.head()

string_list_data  Time of Follow  \
0  [{'href': 'https://www.instagram.com/k1ng_vert...      1725887506   
1  [{'href': 'https://www.instagram.com/swapafr',...      1725679895   
2  [{'href': 'https://www.instagram.com/thankscho...      1725642228   
3  [{'href': 'https://www.instagram.com/n.ieshhaa...      1725386551   
4  [{'href': 'https://www.instagram.com/olarunt',...      1725262958   

  Account Name                       Link to Account  
0   k1ng_vert_  https://www.instagram.com/k1ng_vert_  
1      swapafr     https://www.instagram.com/swapafr  
2   thankschow  https://www.instagram.com/thankschow  
3    n.ieshhaa   https://www.instagram.com/n.ieshhaa  
4      olarunt     https://www.instagram.com/olarunt

In [429]:
del df5["string_list_data"]
df5.head(10)

Time of Follow      Account Name  \
0      1725887506        k1ng_vert_   
1      1725679895           swapafr   
2      1725642228        thankschow   
3      1725386551         n.ieshhaa   
4      1725262958           olarunt   
5      1724616579     phillygyal.__   
6      1724426473  nevergoindigital   
7      1724284995    iwantnishaa216   
8      1724256372   foenemcreations   
9      1724060506            i1_7.c   

                              Link to Account  
0        https://www.instagram.com/k1ng_vert_  
1           https://www.instagram.com/swapafr  
2        https://www.instagram.com/thankschow  
3         https://www.instagram.com/n.ieshhaa  
4           https://www.instagram.com/olarunt  
5     https://www.instagram.com/phillygyal.__  
6  https://www.instagram.com/nevergoindigital  
7    https://www.instagram.com/iwantnishaa216  
8   https://www.instagram.com/foenemcreations  
9            https://www.instagram.com/i1_7.c

Finally got the dataframe looking nice and easy to understand this was one of my main goals because an easy to understand table makes the entire project easy to understand.

In [463]:
df5.tail(10)

Time of Follow          Account Name  \
586      1695621942               ochoov1   
587      1695348504             lysaswrld   
588      1695332576           ralanstyles   
589      1695132859        kaylaa.andrea_   
590      1695092975               d1.quan   
591      1694717860  prototype.collective   
592      1694703421        westonbergmann   
593      1694703375              itsravyn   
594      1694599323            beatsby_jt   
595      1694535395                brezdn   

                                    Link to Account  
586               https://www.instagram.com/ochoov1  
587             https://www.instagram.com/lysaswrld  
588           https://www.instagram.com/ralanstyles  
589        https://www.instagram.com/kaylaa.andrea_  
590               https://www.instagram.com/d1.quan  
591  https://www.instagram.com/prototype.collective  
592        https://www.instagram.com/westonbergmann  
593              https://www.instagram.com/itsravyn  
594            https://www.instagram.com/beatsby_jt  
595                https://www.instagram.com/brezdn

I output the tail of the dataframe just to double check how many rows there were and also to see the time of follow for the oldest entry so that I can add that and the newest entry and then divide by two to have an "average" but I am more so looking at it as a middle point. My hypothesis is that I have been more active following people recently so if I can see how many rows or entries come more recently than the middle point and compare that to how many entries there are as a whole I can have data and evidence to support my hypothesis.

In [477]:
len(df5[df5['Time of Follow'] > 1710211450])

358

There are 595 entires or follows in this dataframe and 358 of those come more recent than the middle point which is 1710211450. 61% of my follows have come more recently than the middle point which should be more than normal one would think about 50% so my hypothesis is correct. I had a lot of fun with this project I would have had more fun and felt better about it if I had more knowledge on pandas or python in general. I wanted to analyze my message times and see if it was reflective of my work and school schedule but I simply do not know enough to execute that correctly. I would like to tackle that subject and perhaps use a graph as I get more knowledge and more comfortable with the class.